hsqldb.te

```diff
@@ -0,0 +1,59 @@
+policy_module(hsqldb, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type hsqldb_t;
+type hsqldb_exec_t;
+init_daemon_domain(hsqldb_t, hsqldb_exec_t)
+
+type hsqldb_tmp_t;
+files_tmp_file(hsqldb_tmp_t)
+
+type hsqldb_var_lib_t;
+files_type(hsqldb_var_lib_t)
+
+type hsqldb_unit_file_t;
+systemd_unit_file(hsqldb_unit_file_t)
+
+########################################
+#
+# hsqldb local policy 
+# 
+
+allow hsqldb_t self:process execmem;
+
+allow hsqldb_t self:fifo_file rw_fifo_file_perms;
+allow hsqldb_t self:stream_socket_class_set create_stream_socket_perms;
+
+manage_dirs_pattern(hsqldb_t, hsqldb_tmp_t, hsqldb_tmp_t)
+manage_files_pattern(hsqldb_t, hsqldb_tmp_t, hsqldb_tmp_t)
+manage_lnk_files_pattern(hsqldb_t, hsqldb_tmp_t, hsqldb_tmp_t)
+files_tmp_filetrans(hsqldb_t, hsqldb_tmp_t, { dir file lnk_file })
+
+manage_dirs_pattern(hsqldb_t, hsqldb_var_lib_t, hsqldb_var_lib_t)
+manage_files_pattern(hsqldb_t, hsqldb_var_lib_t, hsqldb_var_lib_t)
+manage_lnk_files_pattern(hsqldb_t, hsqldb_var_lib_t, hsqldb_var_lib_t)
+create_sock_files_pattern(hsqldb_t, hsqldb_var_lib_t, hsqldb_var_lib_t)
+files_var_lib_filetrans(hsqldb_t, hsqldb_var_lib_t, { dir })
+
+kernel_read_system_state(hsqldb_t)
+kernel_read_network_state(hsqldb_t)
+
+corecmd_exec_bin(hsqldb_t)
+
+corenet_tcp_bind_generic_node(hsqldb_t)
+corenet_tcp_bind_tor_port(hsqldb_t)
+corenet_tcp_connect_tor_port(hsqldb_t)
+
+dev_list_sysfs(hsqldb_t)
+
+dev_read_urand(hsqldb_t)
+dev_read_rand(hsqldb_t)
+
+
+auth_use_nsswitch(hsqldb_t)
```

Comment:
On Thu, Aug 13, 2015 at 05:40:48AM -0700, Lukas Vrabec wrote:
> > +kernel_read_system_state(hsqldb_t)
> > +kernel_read_network_state(hsqldb_t)
> > +
> > +corecmd_exec_bin(hsqldb_t)
> > +
> > +corenet_tcp_bind_generic_node(hsqldb_t)
> > +corenet_tcp_bind_tor_port(hsqldb_t)
> > +corenet_tcp_connect_tor_port(hsqldb_t)
> > +
> > +dev_list_sysfs(hsqldb_t)
> > +
> > +dev_read_urand(hsqldb_t)
> > +dev_read_rand(hsqldb_t)
> > +
> > +
> > +auth_use_nsswitch(hsqldb_t)
> 
> Follow alphabetical order please ;) 

The snippet above is actually close to proper.

The auth module is not part of the kernel layer and should be below any kernel layer interface calls as per:

https://github.com/TresysTechnology/refpolicy/wiki/StyleGuide

In particular:

1. calls to kernel layer modules, sorted in alphabetical order (exception: kernel module is first)
2. calls to system layer modules, sorted in alphabetical order
3. calls to all other modules, sorted in alphabetical order


> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/fedora-selinux/selinux-policy/pull/30/files#r36969159

-- 
02DFF788
4D30 903A 1CF3 B756 FB48  1514 3148 83A2 02DF F788
http://keys.gnupg.net/pks/lookup?op=vindex&search=0x314883A202DFF788
Dominick Grift